consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/blockcreation/PayloadIdentifier.java

```diff
@@ -71,23 +72,26 @@ public static PayloadIdentifier forPayloadParams(
       final Bytes32 prevRandao,
       final Address feeRecipient,
       final Optional<List<Withdrawal>> withdrawals,
-      final Optional<Bytes32> parentBeaconBlockRoot) {
+      final Optional<Bytes32> parentBeaconBlockRoot,
+      final Optional<UInt64> targetBlobCount) {
 
     return new PayloadIdentifier(
         timestamp
             ^ ((long) parentHash.toHexString().hashCode()) << 8
             ^ ((long) prevRandao.toHexString().hashCode()) << 16
             ^ ((long) feeRecipient.toHexString().hashCode()) << 24
             ^ (long)
-                withdrawals
-                    .map(
-                        ws ->
-                            ws.stream()
-                                .sorted(Comparator.comparing(Withdrawal::getIndex))
-                                .map(Withdrawal::hashCode)
-                                .reduce(1, (a, b) -> a ^ (b * 31)))
-                    .orElse(0)
-            ^ ((long) parentBeaconBlockRoot.hashCode()) << 40);
+                    withdrawals
+                        .map(
+                            ws ->
+                                ws.stream()
+                                    .sorted(Comparator.comparing(Withdrawal::getIndex))
+                                    .map(Withdrawal::hashCode)
+                                    .reduce(1, (a, b) -> a ^ (b * 31)))
+                        .orElse(0)
+                << 32
```

Comment:
Note, added a shift for withdrawals to keep the hashcodes evenly distributed by using the same pattern as the other fields.

Reply:
is it worth extracting this to a method? it's a bit hard to read indented over so many lines